teuthology/orchestra/remote.py

```diff
@@ -599,6 +598,22 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
+    def write_temp_file(self, **kwargs):
+        """
+        This is a convenience method that call write_file() but without sudo.
+        Reason for excluding sudo is that files in /tmp can not be written by
+        any user (including the root user) besides the file owner.
+
+        Takes arguments of write_file() and of mktemp().
+        """
+        kwargs['sudo'] = False
+
+        if not kwargs.get('path'):
+            kwargs['path'] = self.mktemp(kwargs.pop('suffix',  None),
```

Comment:
Instead of adding another method, could you please extend `mktemp` method with `data` argument, and if it not None, call remote_write.

Reply:
Thanks for the review.

> Instead of adding another method, could you please extend `mktemp` method with `data` argument,

Okay, I get this much...

> and if it not None, call remote_write.

but I don't get this part. By remote_write do you mean `remote.Remote.write_file()`?

Also, just adding `data` as parameter to `Remote.mktemp` won't be enough in case someone wants to wanted to append instead of overwrite, set owner, perm modes, etc. too on the file. So, will it be clean and simple if mktemp accepted all arguments of `write_file()` besides `suffix` and `parentdir`?

I like the idea of extending mktemp(), I am just wondering if there is a case that extended mktemp() can't cover but two separate methods can.

Reply:
there is no sense at all in 'append' feature when creating a temporary file, because the whole idea of the _mktemp_ is always to create a new file with distinct name. The _mktemp_ method is supposed to be used in similar way as system `mktemp`, which is not likely used to have another owner than invoking user and mode is supposed to be 0600. If we really have use cases when we need other modes and owner, maybe we can extend mktemp, with additional parameters, but I don't think we need this.

Reply:
Extended mktemp() and deleted write_temp_file().